lib/srv/sess.go

```diff
@@ -1092,6 +1060,39 @@ func (s *session) startInteractive(ch ssh.Channel, ctx *ServerContext, tempUser
 	// Start a heartbeat that marks this session as active with current members
 	// of party in the backend.
 	go s.heartbeat(ctx)
+
+	// wait for exec.Cmd (or receipt of "exit-status" for a forwarding node),
+	// once it is received wait for the io.Copy above to finish, then broadcast
+	// the "exit-status" to the client.
+	go func() {
+		result, err := s.term.Wait()
+		if err != nil {
+			ctx.Errorf("Received error waiting for the interactive session %v to finish: %v.", s.id, err)
+		}
+
+		// wait for copying from the pty to be complete or a timeout before
```

Comment:
If we got an error above should we still be waiting here?

Reply:
I think so, wait can fail for a number of reasons but at least some of them still leave the terminal data valid so I think it makes sense to ensure it's all been copied.